meta/main.yml

```diff
@@ -16,9 +16,6 @@ galaxy_info:
     - 7
   categories:
   - system
-dependencies: []
-  # List your role dependencies here, one per line. Only
-  # dependencies available via galaxy should be listed here.
-  # Be sure to remove the '[]' above if you add dependencies
-  # to this list.
-  
+dependencies:
+  - role: tor
+    when: "enable_onion_support == True"
```

Comment:
So after a test, when seems to work fine without the == True, provided we have a default set, or it complain about undefined variable. It also do the conversion from 0 and "" to False irrespective of the type.